src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -202,18 +204,44 @@ ss::future<> service::configure() {
       _ctx.smp_sg, [has_ephemeral_credentials](service& s) {
           s._has_ephemeral_credentials = has_ephemeral_credentials;
       });
-    co_await _controller->get_security_frontend().local().create_acls(
-      {security::acl_binding{
-        security::resource_pattern{
-          security::resource_type::topic,
-          model::schema_registry_internal_tp.topic,
-          security::pattern_type::literal},
-        security::acl_entry{
-          principal,
-          security::acl_host::wildcard_host(),
-          security::acl_operation::all,
-          security::acl_permission::allow}}},
-      5s);
+
+    std::vector<cluster::errc> err_vec;
+    // This could be any non-success errc
+    err_vec.push_back(cluster::errc::timeout);
+
+    co_await ss::do_until(
+      [&err_vec] { return err_vec[0] == cluster::errc::success; },
+      [this, &err_vec] {
+          return _controller->get_security_frontend()
+            .local()
+            .create_acls(
+              {security::acl_binding{
+                security::resource_pattern{
+                  security::resource_type::topic,
+                  model::schema_registry_internal_tp.topic,
+                  security::pattern_type::literal},
+                security::acl_entry{
+                  principal,
+                  security::acl_host::wildcard_host(),
+                  security::acl_operation::all,
+                  security::acl_permission::allow}}},
+              5s)
+            .then([&err_vec](std::vector<cluster::errc> ev) {
+                err_vec = std::move(ev);
+                if (err_vec[0] != cluster::errc::success) {
+                    vlog(
+                      plog.warn,
+                      "Failed creating ACLs, User {}, err {} - {}",
+                      principal,
+                      err_vec[0],
+                      cluster::make_error_code(err_vec[0]).message());
+                    return ss::sleep(create_acls_backoff);
```

Comment:
Is this OK to block startup? My thought is that if SR is waiting on something such as a leader for the controller, then everything else in the cluster is also waiting for the same thing. Therefore, this sleep should be OK right?